GeoLib/Surface.h

```diff
@@ -36,31 +32,36 @@ class SurfaceGrid;
  * to a vector of (pointers to) points (_sfc_pnts) and a vector that stores
  * the Triangles consisting of points from _sfc_pnts.
  * */
-class Surface : public GeoObject
+class Surface final : public GeoObject
 {
 public:
-    Surface(const std::vector<Point*> &pnt_vec);
-    virtual ~Surface ();
+    explicit Surface(const std::vector<Point*>& pnt_vec);
+    explicit Surface(Surface const& src);
+    ~Surface();
 
-    /// return a geometry type
-    virtual GEOTYPE getGeoType() const {return GEOTYPE::SURFACE;}
+    Surface(Surface && src) = delete;
+    Surface& operator=(Surface const& src) = delete;
```

Comment:
I implemented the copy constructor since @rinkk needs it. The only reason the move constructor and the assignment and assignment move operators are not implemented is that they are not needed until now.